frontend-react/src/hooks/network/UseEndpoint.test.ts

```diff
@@ -0,0 +1,82 @@
+import { act, renderHook } from "@testing-library/react-hooks";
+import { rest } from "msw";
+import { setupServer } from "msw/node";
+
+import { mockSessionContext } from "../../contexts/__mocks__/SessionContext";
+import {
+    MembershipController,
+    MembershipSettings,
+} from "../UseOktaMemberships";
+import { SessionController } from "../UseSessionStorage";
+import { MyApi, MyApiItem } from "../../network/api/test-tools/MockApi";
+import { SimpleError } from "../../utils/UsefulTypes";
+
+import useEndpoint, { passesObjCompare } from "./UseEndpoint";
+
+const dummyArrayReturn = [new MyApiItem("test1"), new MyApiItem("test2")];
+const handlers = [
+    /* Returns a list of two fake api items */
+    rest.get(
+        "https://test.prime.cdc.gov/api/test/test/:id",
+        (_req, res, ctx) => {
+            return res(ctx.status(200), ctx.json(new MyApiItem("test")));
+        }
+    ),
+    rest.get("https://test.prime.cdc.gov/api/test/test", (_req, res, ctx) => {
+        return res(ctx.status(200), ctx.json(dummyArrayReturn));
+    }),
+];
+
+const mockSession = {
+    oktaToken: {
+        accessToken: "TOKEN",
+    },
+    memberships: {
+        state: {
+            active: {
+                parsedName: "ORGANIZATION",
+            } as MembershipSettings,
+        },
+    } as MembershipController,
+    store: {} as SessionController,
+};
+
+/* TEST SERVER TO USE IN `.test.ts` FILES */
+const testServer = setupServer(...handlers);
+
+describe("useEndpoint", () => {
```

Comment:
maybe also a test to see what happens when the return value returns false from `passesObjCompare`?